src/lib/hooks/useMessageTemplateUtils.ts

```diff
@@ -0,0 +1,176 @@
+  MARK_ERROR_WAITING_TEMPLATE_KEY,
+} = APP_INFO_ACTIONS;
+
+export default function useMessageTemplateUtils({
+  sdk,
+  logger,
+  appInfoStore,
+  appInfoDispatcher,
+}): UseMessageTemplateUtilsWrapper {
+  const messageTemplatesInfo: MessageTemplatesInfo | undefined = appInfoStore?.messageTemplatesInfo;
+
+  const getCachedTemplate = (key: string): ProcessedMessageTemplate | null => {
+    if (!messageTemplatesInfo) return null;
+
+    let cachedTemplate: ProcessedMessageTemplate | null = null;
+    const cachedMessageTemplates: Record<string, ProcessedMessageTemplate> | null = messageTemplatesInfo?.templatesMap ?? null;
+    if (cachedMessageTemplates) {
+      cachedTemplate = cachedMessageTemplates[key] ?? null;
+    }
+    return cachedTemplate;
+  };
+
+  /**
+   * Fetches a single message template by given key and then
+   * returns processed template for updating templates info in global state.
+   * If no such templates exists (error) or any error occurs in response, return null.
+   */
+
+  const fetchAllMessageTemplates = async (readySdk: SendbirdChat): Promise<SendbirdMessageTemplate[]> => {
+    let hasMore = true;
+    let paginationToken = null;
+    const fetchedTemplates: SendbirdMessageTemplate[] = [];
+
+    while (hasMore) {
+      /**
+       * RFC doc:
+       * https://sendbird.atlassian.net/wiki/spaces/PLAT/pages/2254405651/RFC+Message+Template#%5BAPI%5D-List-message-templates
+       */
+      const res: MessageTemplateListResult = await readySdk!.message.getMessageTemplatesByToken(
+        paginationToken,
+        { limit: MESSAGE_TEMPLATES_FETCH_LIMIT },
+      );
+      hasMore = res.hasMore;
+      paginationToken = res.token;
+      res.templates.forEach((messageTemplate) => {
+        fetchedTemplates.push(JSON.parse(messageTemplate.template));
+      });
+    }
+    return fetchedTemplates;
+  };
+
+  const initializeMessageTemplatesInfo = async (readySdk: SendbirdChat): Promise<void> => {
+    const sdkMessageTemplateToken = readySdk!.appInfo?.messageTemplateInfo.token;
+
+    /**
+     * no sdkMessageTemplateToken => no templates => clear cached
+     */
+    if (!sdkMessageTemplateToken) {
+      localStorage.removeItem(CACHED_MESSAGE_TEMPLATES_TOKEN_KEY);
+      localStorage.removeItem(CACHED_MESSAGE_TEMPLATES_KEY);
+      return;
+    }
+    /**
+     * Given the following cases:
+     * 1. non-null sdkMessageTemplateToken => templates exist
+     * 2. no cached token or cached token is outdated => first fetch or outdated cache
+     *
+     * If both 1 and 2, fetch all templates and upsert to cache.
+     * If cached token is not outdated, use cached templates.
+     */
+    const cachedMessageTemplatesToken: string | null = localStorage.getItem(CACHED_MESSAGE_TEMPLATES_TOKEN_KEY);
+    const cachedMessageTemplates: string | null = localStorage.getItem(CACHED_MESSAGE_TEMPLATES_KEY);
+    if (
+      !cachedMessageTemplatesToken
+      || cachedMessageTemplatesToken !== sdkMessageTemplateToken!
```

Comment:
```suggestion
      || cachedMessageTemplatesToken !== sdkMessageTemplateToken
```

Reply:
I think its safer to put ! if we expect non null/undefined object no?